test/constructRoutes.test.js

```diff
@@ -1,16 +1,82 @@
 import { inBrowser } from "../src/environment-helpers.js";
 import { constructRoutes } from "../src/single-spa-layout.js";
+import fs from "fs";
+import path from "path";
+import { screen } from "@testing-library/dom";
+import parse5 from "parse5";
 
 jest.spyOn(console, "warn");
 
 describe("constructRoutes", () => {
-  let registerApplication = jest.fn();
-
   beforeEach(() => {
-    registerApplication.mockReset();
     console.warn.mockReset();
   });
 
+  describe(`HTML parsing`, () => {
+    it(`can parse a medium complexity HTML routes definition`, () => {
+      const document = parseHTML(
+        fs.readFileSync(
+          path.resolve(__dirname, "./fixtures/medium.html"),
+          "utf-8"
+        )
+      );
+      // In browser we can use querySelector, otherwise the more manual lookup
+      const routerElement = inBrowser
+        ? document.querySelector("single-spa-router")
+        : document.childNodes[1].childNodes[2].childNodes[1];
+      const routes = constructRoutes(routerElement);
+
+      expect(routes.mode).toBe("history");
+      expect(routes.base).toBe("/");
+      expect(routes.routes.length).toBe(5);
+      expect(routes.sourceElement).toBe(routerElement);
+
+      expect(routes.routes[0].type).toBe("application");
+      expect(routes.routes[0].name).toBe("@org/navbar");
+      expect(routes.routes[0].containerEl.nodeName.toLowerCase()).toBe("body");
+
+      expect(routes.routes[1].type).toBe("route");
```

Comment:
These get extremely specific any reason why you didn't use snapshots like you did [elsewhere](https://github.com/single-spa/single-spa-layout/blob/master/test/browser-only/__snapshots__/constructLayoutEngine-browser.test.js.snap)

Reply:
good point - i'll try changing it over. Might need to move this into separate files for node and browser so the snapshots don't collide

Reply:
I tried switching it to snapshots and realized why I didn't do that. `containerEl` is a dom element and that gets serialized in weird ways.

Reply:
Might switch this if we change container els. But not worth it right now